python/semantic_kernel/agents/agent.py

```diff
@@ -18,11 +18,66 @@
 from semantic_kernel.prompt_template.kernel_prompt_template import KernelPromptTemplate
 from semantic_kernel.prompt_template.prompt_template_base import PromptTemplateBase
 from semantic_kernel.prompt_template.prompt_template_config import PromptTemplateConfig
+from semantic_kernel.utils.feature_stage_decorator import release_candidate
 from semantic_kernel.utils.naming import generate_random_ascii_name
 from semantic_kernel.utils.validation import AGENT_NAME_REGEX
 
 logger: logging.Logger = logging.getLogger(__name__)
 
+TMessage = TypeVar("TMessage", bound=ChatMessageContent | StreamingChatMessageContent)
```

Comment:
StreamingChatMessageContent is a subclass of ChatMessageContent, so might not even need this...